src/provider.c

```diff
@@ -1388,6 +1398,37 @@ static struct p11prov_cfg_names {
     { "pkcs11-module-quirks" },
     { "pkcs11-module-cache-sessions" },
     { "pkcs11-module-encode-provider-uri-to-pem" },
+    { "pkcs11-module-block-operations" },
+};
+
+static struct p11prov_block_ops_names {
+    const char *name;
+    size_t op_num;
```

Comment:
nit: Do you want to include op_num in the struct given it is monotonically increasing? If we can assume C99, it is also possible to use designator in the array initializer:
```c
const char *p11prov_block_ops_names[OSSL_OP__HIGHEST + 1] = {
    [OSSL_OP_DIGEST] = "digest",
    [OSSL_OP_CIPHER] = "cipher",
    ...
};
```